java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -42,11 +44,152 @@
+      this.validity = validity;
+      this.type = type;
+      this.rows = rows;
+      this.nullCount = nullCount;
+      this.nestedChildren = nestedChildren;
+    }
+
+    @Override
+    public void close() {
+      for (NestedHostColumnVector child : nestedChildren) {
+        if (child != null) {
+          child.close();
+        }
+      }
+      if (data != null) {
+        data.close();
+      }
+      if (validity != null) {
+        validity.close();
+      }
+      if (offsets != null) {
+        offsets.close();
+      }
+    }
+
+    HostMemoryBuffer getData() {
+      return data;
+    }
+
+    HostMemoryBuffer getValidity() {
+      return validity;
+    }
+
+    DType getType() {
+      return type;
+    }
+
+    Optional<Long> getNullCount() {
+      return nullCount;
+    }
+
+    List<NestedHostColumnVector> getNestedChildren() {
+      return nestedChildren;
+    }
+
+    public HostMemoryBuffer getOffsets() {
+      return offsets;
+    }
+
+    public long getRows() {
+      return rows;
+    }
+
+
+    private Object getElement(int rowIndex) {
+      if (type == DType.LIST) {
+        List retList = new ArrayList();
+        int start = offsets.getInt(rowIndex * DType.INT32.getSizeInBytes());
+        int end = offsets.getInt((rowIndex + 1) * DType.INT32.getSizeInBytes());
+        for (int j = start; j < end; j++) {
+          retList.add(nestedChildren.get(0).getElement(j));
+        }
+        return retList;
+      } else if (type == DType.STRING) {
+        int start = offsets.getInt(rowIndex * DType.INT32.getSizeInBytes());
+        int end = offsets.getInt((rowIndex + 1) * DType.INT32.getSizeInBytes());
+        int size = end - start;
+        byte[] rawData = new byte[size];
+        if (size > 0) {
+          data.getBytes(rawData, 0, start, size);
+          return new String(rawData);
+        } else if (isNull(rowIndex)) {
+          return null;
+        } else {
+          return new String();
+        }
+      } else {
+        if (isNull(rowIndex)) {
+          return null;
+        }
+        int start = rowIndex * type.getSizeInBytes();
+        return readValue(start);
+      }
+    }
+
+    public boolean isNull(long index) {
+      assert (index >= 0 && index < rows) : "index is out of range 0 <= " + index + " < " + rows;
+      if (validity != null) {
+        return BitVectorHelper.isNull(validity, index);
+      }
+      return false;
+    }
+
+    private Object readValue(int index){
+      assert index < rows * type.getSizeInBytes();
+      switch (type) {
+        case INT32: return data.getInt(index);
+        case INT64: return data.getLong(index);
+        case FLOAT32: return data.getFloat(index);
+        case FLOAT64: return data.getDouble(index);
+        case INT8: return data.getByte(index);
+        case INT16: return data.getShort(index);
+        case BOOL8: return data.getBoolean(index);
+        default: throw new UnsupportedOperationException("Do not support " + type);
```

Comment:
Are Strings not supported by lists?

Reply:
Strings are Supported per code in `getElement` and I have added tests as well to show how List of strings work. Let me know if you have any questions.